internal/policies/mount/adsys_mount/src/main.rs

```diff
@@ -0,0 +1,235 @@
+    let args = Args::parse();
+
+    // Creates the logger and sets its level to Debug.
+    if let Ok(()) = log::set_logger(&Logger {}) {
+        log::set_max_level(log::LevelFilter::Debug);
+    }
+
+    debug!("Mounting entries listed in {}", args.mounts_file);
+
+    let parsed_entries = match parse_entries(&args.mounts_file) {
+        Ok(v) => v,
+        Err(e) => {
+            error!("Error when parsing entries: {}", e);
+            return Err(AdsysMountError::ParseError);
+        }
+    };
+
+    // Setting up the channel used for communication between the mount operations and the main function.
+    let g_ctx = glib::MainContext::default();
+    let (tx, rx) = glib::MainContext::channel(glib::PRIORITY_DEFAULT);
+
+    // Grabs the ammount of mounts to be done before passing the ownership of parsed_entries.
+    let mut mounts_left = parsed_entries.len();
+
+    for entry in parsed_entries {
+        handle_mount(entry, tx.clone());
+    }
+
+    // Sets the main loop glib to be used by the mounts
+    let g_loop = glib::MainLoop::new(Some(&g_ctx), false);
+
+    // Creates a mutex to handle the exit status
+    let mu: Arc<Mutex<Vec<Msg>>> = Arc::new(Mutex::new(Vec::new()));
+
+    // Clones the variables that are going to be moved into the closure.
+    let g_loop_clone = g_loop.clone();
+    let mu_clone = Arc::clone(&mu);
+
+    // Attaches the receiver to the main context, along with a closure that is called everytime there is a new message in the channel.
+    rx.attach(Some(&g_ctx), move |x| {
+        match x.status {
+            MountStatus::Done => debug!("Mounting of {} was successful", x.path),
+            MountStatus::Error(_) => {
+                warn!("Failed when mounting {}", x.path);
+                mu_clone.lock().unwrap().push(x);
+            }
+            _ => {}
+        };
+        mounts_left -= 1;
+        glib::Continue(match mounts_left {
+            0 => {
+                // Ends the main loop if there are no more mounts left.
+                g_loop_clone.quit();
+                false
+            }
+            _ => true,
+        })
+    });
+
+    g_loop.run();
+
+    // Evaluates the arc content to check if at least one operation failed.
+    let mut had_error = false;
+    let errors = mu.lock().unwrap();
+    if errors.len() != 0 {
+        for err in errors.iter() {
+            if let MountStatus::Error(e) = &err.status {
+                warn!("Mount process for {} failed: {}", err.path, e);
+
+                // Ensures that the function will not error out if the location was already mounted.
+                if !e.matches(gio::IOErrorEnum::AlreadyMounted) {
+                    had_error = true;
+                }
+            }
+        }
+    }
+
+    if had_error {
+        return Err(AdsysMountError::MountError);
+    }
+    Ok(())
+}
+
+/// Reads the file and parses the mount points listed in it.
+fn parse_entries(path: &String) -> Result<Vec<MountEntry>, std::io::Error> {
```

Comment:
I think you can safely use a &str, thanks to implicit coerxion, you don’t even need to typecast.

It will make some change where you can pass directly the slice reference in the test.